tests/queries/0_stateless/01731_async_task_queue_wait.sh

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+
+# regression for 'Empty task was returned from async task queue' during query
+# cancellation with async_socket_for_remote=1 (that ignores
+# max_distributed_connections)
+timeout 5s ${CLICKHOUSE_CLIENT} --max_distributed_connections=1 --format Null -q "select * from remote('127.{2..11}', view(select * from numbers(1e9))) group by number format Null"
```

Comment:
Your query does not reproduce the bug for me.

This one reproduces it almost always
```
timeout --signal=SIGINT 1 clickhouse client --max_distributed_connections=1 --max_block_size=2  --interactive_delay=900000 -q "select x  from remote('127.{2,3}', view(select number + sleep(0.3) as x from numbers(16))) settings max_block_size = 2" | grep "Empty task was returned from async task queue"
Received exception from server (version 21.3.1):
Code: 49. DB::Exception: Received from localhost:9000. DB::Exception: Empty task was returned from async task queue.
```

Reply:
@KochetovNicolai I've just tried you query, and it does not always reproduce the problem for me (but mine variant also not 100%, maybe this is because right now machine is under load...)

Also [ANTLR parser does not support view](https://clickhouse-test-reports.s3.yandex.net/20881/9c01869090e873603b3bb7ec1cd17fbcf264bc4f/functional_stateless_tests_(antlr_debug).html#fail1), and seems that we can simply remove wrapping into view with your variant of the query.

Reply:
@KochetovNicolai I've fixed these (and plus bash syntax), can you also squash before merge please

Also I've merged upstream/master into this PR to catch possible uncaught exception cases with diagnostics with gdb